Gems/Atom/RPI/Code/Source/RPI.Builders/Pass/PassBuilder.cpp

```diff
@@ -65,35 +65,45 @@ namespace AZ
             m_isShuttingDown = true;
         }
 
-        void PassBuilder::CreateJobs(const AssetBuilderSDK::CreateJobsRequest& request, AssetBuilderSDK::CreateJobsResponse& response) const
+        // --- Code related to dependency shader asset handling ---
+
+        // Helper class to pass parameters to the AddDependency and FindReferencedAssets functions below
+        struct FindPassReferenceAssetParams
         {
-            if (m_isShuttingDown)
-            {
-                response.m_result = AssetBuilderSDK::CreateJobsResultCode::ShuttingDown;
-                return;
-            }
+            void* passAssetObject;
+            Uuid passAssetUuid;
+            SerializeContext* serializeContext;
+            AZStd::string_view passAssetSourceFile;     // File path of the pass asset
+            AZStd::string_view dependencySourceFile;    // File pass of the asset the pass asset depends on
+            const char* jobKey;                         // Job key for adding job dependency
+        };
+
+        // Helper function to get a file reference and create a corresponding job dependency
+        void AddDependency(FindPassReferenceAssetParams& params, AssetBuilderSDK::JobDescriptor* job)
+        {
+            AZStd::string_view& file = params.dependencySourceFile;
+            AZ::Data::AssetInfo sourceInfo;
+            AZStd::string watchFolder;
+            bool fileFound = false;
+            AzToolsFramework::AssetSystemRequestBus::BroadcastResult(fileFound, &AzToolsFramework::AssetSystem::AssetSystemRequest::GetSourceInfoBySourcePath, file.data(), sourceInfo, watchFolder);
 
-            for (const AssetBuilderSDK::PlatformInfo& platformInfo : request.m_enabledPlatforms)
+            if (fileFound)
             {
-                AssetBuilderSDK::JobDescriptor job;
-                job.m_jobKey = PassBuilderJobKey;
-                job.SetPlatformIdentifier(platformInfo.m_identifier.c_str());
-
-                // Passes are a critical part of the rendering system
-                job.m_critical = true;
-
-                response.m_createJobOutputs.push_back(job);
+                AssetBuilderSDK::JobDependency jobDependency;
+                jobDependency.m_jobKey = params.jobKey;
+                jobDependency.m_type = AssetBuilderSDK::JobDependencyType::Order;
+                jobDependency.m_sourceFile.m_sourceFileDependencyPath = file;
+                job->m_jobDependencyList.push_back(jobDependency);
+                AZ_TracePrintf(PassBuilderName, "Creating job dependency on file [%s] \n", file.data());
             }
```

Comment:
is it OK if the file isn't found? Should there be a warning or error here if it can't find the dependency?